app/gui/src/ide/integration/project.rs

```diff
@@ -770,9 +770,12 @@ impl Model {
     /// Reload whole displayed content to be up to date with module state.
     pub fn refresh_graph_view(&self) -> FallibleResult {
         info!(self.logger, "Refreshing the graph view.");
+        let performance_handle = performance_logger::start_interval_task("refresh_graph_view");
         let connections_info = self.graph.connections()?;
         self.refresh_node_views(&connections_info, true)?;
         self.refresh_connection_views(connections_info.connections)?;
+        let _ = performance_handle.measure();
+
```

Comment:
This is not the API that was designed in the tech doc. Please change it to the designed API. If you want to discuss the designed API, let's make the discussion in the doc in the first place.